providers/snapchatads/utils.go

```diff
@@ -2,13 +2,44 @@ package snapchatads
 
 import (
 	"github.com/amp-labs/connectors/common/urlbuilder"
+	"github.com/amp-labs/connectors/internal/datautils"
+)
+
+// The endpoints shared id(organization Id) in the url path.
+var endpointsWithSharedId = datautils.NewSet( //nolint:gochecknoglobals
+	"fundingsources",
+	"billingcenters",
+	"transactions",
+	"adaccounts",
+	"members",
+	"roles",
 )
 
 func (c *Connector) constructURL(objName string) (*urlbuilder.URL, error) {
-	url, err := urlbuilder.New(c.ProviderInfo().BaseURL, apiVersion, "organizations", c.organizationId, objName)
-	if err != nil {
-		return nil, err
+	var prefix string
+
+	switch objName {
+	case "age_group", "gender", "languages", "advanced_demographics":
+		prefix = "targeting/demographics/"
```

Comment:
we shouldn't be mapping names (as we've recently discussed in other PRs), let's have these as `targeting/demographics/languages` and so on